### DISCUSSION TITLE
Korean shift breaks the character in GTK 3/4 apps only

### DISCUSSION THREAD

**Discussion by HubKing** [fcitx/fcitx5#1156] [Q&A] [answered]
Arch Linux. After yesterday's updates, in GTK 3 or 4 apps such as FireFox, if I press shift, it breaks the character. For example, to input 갔, if I press r, k, and shift + t, it becomes 가ㅆ. So, as a temporary measure, I am pressing r, k, t, and t to input 갔. This problem does not happen on QT apps.

GTK4 Text Editor
[Screencast_20241013_061420.webm](https://github.com/user-attachments/assets/eeb6416a-fb7b-4bde-981b-0cb534d4e001)

QT6 KWrite
[Screencast_20241013_061706.webm](https://github.com/user-attachments/assets/9509ae40-5ae5-4cfe-9dc6-cc1bf33ed61d)

Does anyone else have this problem?

**Comment by wengxt** [comment]
I can't reproduce this with plasma 6.1.5. I believe this is a kwin issue.

kwin 6.2 added commit preedit on certain event. It seems somehow shift key press triggers this hebahvior.

**Reply by HubKing** [reply]
Yeah, I'm on Plasma 6.2. Is there a workaround?

**Reply by HubKing** [reply]
I mean, for double consonants, I can just type it twice without using the shift, but I don't know how to type nested vowels like ㅖ without the shift.

**Reply by wengxt** [reply]
> Yeah, I'm on Plasma 6.2. Is there a workaround?

No, but you can try my patch on to kwin https://invent.kde.org/plasma/kwin/-/merge_requests/6605

Downgrade to 6.1.5 might work if it doesn't crash (kwin has dependency on other packages so downgrade only kwin is not guaranteed to work).

**Reply by wengxt** [reply, marked as answer]
I just merged the fix to 6.2 and master branch. So next minor version would contain the fix.

Oh, btw, an easy workaround is to use gtk im module by setting GTK_IM_MODULE=fcitx .

**Reply by HubKing** [reply]
Thanks, but where to add `GTK_IM_MODULE=fcitx` ? I added it to `~/.zshrc` and rebooted, but that did not fix the problem.

**Reply by HubKing** [reply]
Setting it in the `~/.config/environement.d/config.conf` (and rebooting) worked. However, I see this pop-up when starting the desktop.

![image](https://github.com/user-attachments/assets/b1ce24a3-90d5-4ad7-8203-c5330d30249e)

**Reply by wengxt** [reply]
> Setting it in the `~/.config/environement.d/config.conf` (and rebooting) worked. However, I see this pop-up when starting the desktop.
> 
> ![image](https://private-user-images.githubusercontent.com/10900489/376072905-b1ce24a3-90d5-4ad7-8203-c5330d30249e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4Nzk2NTcsIm5iZiI6MTcyODg3OTM1NywicGF0aCI6Ii8xMDkwMDQ4OS8zNzYwNzI5MDUtYjFjZTI0YTMtOTBkNS00YWQ3LTgyMDMtYzUzMzBkMzAyNDllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDA0MTU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwOWE2ZWQ2YmU4NDVmYmMzOWM5MThmNGY3OWQ2N2NlY2U5YTBkZWU5NjA0NzU1ZTk5ZDg1ZmJjY2Y0ZTc3NjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CAG773CfznivV4FVDaban19myxTjvZ25REmxbB5HYTs)

it's just some advice, not a big deal. just click dont show if you find it annoying.
